### PR TITLE
Fixed compiler warnings

### PR DESCRIPTION
**Pull request by tmiguelf**
fixed warnings
used proper math functions

### CODE REVIEW

**Comment by dmah42** [comment]
sanitizer failures are unrelated, but the formatting should be fixed please.  running clang-format with style "Google" should do it.

**Comment by tmiguelf** [comment]
@dmah42 Thanks for the review, hopefully that should do it.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Allow the source files to find headers in src/
+#Allow the source files to find headers in src /
```
Comment (dmah42) [inline src/CMakeLists.txt:1]: this looks wrong

Comment (tmiguelf) [reply src/CMakeLists.txt]: Reverted it

**Review by dmah42 — APPROVED** [review]

**Comment by dmah42** [comment]
thank you!

**Pull request merged**